### PR TITLE
A structured approach to work with software development tasks

### PR DESCRIPTION
**Pull request by Dunklas**

### CODE REVIEW

File: content/posts/2023-06-01_a-structured-approach-to-complete-software-development-tasks.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+You instantly start to chase new ideas, while forgetting about prior (and possibly important) ideas.
+It might be hard to determine when you're done.
+You might not commit intermediary working states, and have a hard time finding what you've messed up once something is broken.
+
+In this post, I aim to explain how I like to approach tasks in a bit more structured way.
+
+First of all, you need to have a basic understanding the task.
+Read through the description.
+Do some reconnaissance.
+Anything that need to be clarified?
```
Comment (BunnyFiscuit) [inline content/posts/2023-06-01_a-structured-approach-to-complete-software-development-tasks.md:23]: needs*

File: content/posts/2023-06-01_a-structured-approach-to-complete-software-development-tasks.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+If possible, I personally prefer to start with a naive end-to-end implementation of the main chunk of the feature.
+Delaying secondary but important aspects, such as input validation, proper error handling, etc.
+
+Now is an excellent time for a break.
+Stretch your legs, take some fresh air.
+
+Once you're back - Execute!
+Grab the first item from your high level plan.
+Write tests, implement.
+If your mind start to stray towards items further down in your plan, stop!
```
Comment (BunnyFiscuit) [inline content/posts/2023-06-01_a-structured-approach-to-complete-software-development-tasks.md:44]: starts*

File: content/posts/2023-06-01_a-structured-approach-to-complete-software-development-tasks.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+If you get an idea of something important that you've completely forgot to account for in the high level plan - Don't chase that idea right now, simply add a note of it to the bottom of your high level plan.
+If you get stuck on a non-critical aspect of the item you're currently working on - Make it into its own item in the high level plan and deal with it later.
+
+Once you're done with the first item is an excellent time to commit to version control.
+This doesn't necessarily need to be a commit in your main branch - It could be squashed into a larger commit later as well.
+The purpose is to have a clean state before starting with the next item.
+Review your high level plan.
+Put a checkmark next to the item you just completed.
+Do you need to re-prioritize?
+Need to investigate some thing that you recently added?
```
Comment (BunnyFiscuit) [inline content/posts/2023-06-01_a-structured-approach-to-complete-software-development-tasks.md:55]: something*

**Comment by Dunklas** [comment]
Thanks @BunnyFiscuit ! I should get used to run a spellcheck on stuff before opening a PR! :)

**Comment by BunnyFiscuit** [comment]
Haha sorry if I was a bit picky x) Otherwise good blog!

**Review by BunnyFiscuit — DISMISSED** [review]

**Review by BunnyFiscuit — APPROVED** [review]

**Pull request merged**